### DISCUSSION TITLE
Sentry: don't init in dev

### DISCUSSION THREAD

**Discussion by proko** [vercel/next.js#26360] [Help]
Hi there! I must be missing something fundamental here but how should one prevent the sentry webpack plugin from creating new releases in development mode? There used to be a way to run the `init()` only in production but missing this option with the new implementation. 
I'm using the code from the [official example](https://github.com/vercel/next.js/tree/canary/examples/with-sentry).

**Comment by jamesmosier** [comment]
Have you tried setting the [DSN to an empty (or null) value](https://github.com/getsentry/sentry-java/issues/574) which is supposed to disable logging? Inside the `sentry.client.config.js` and `sentry.server.config.js` you could change:

```js
const SENTRY_DSN = process.env.SENTRY_DSN || process.env.NEXT_PUBLIC_SENTRY_DSN
```

to 

```js
const SENTRY_DSN = process.env.NODE_ENV === 'production' ? process.env.SENTRY_DSN || process.env.NEXT_PUBLIC_SENTRY_DSN : ''
```

**Reply by Vadorequest** [reply]
Sometimes, having Sentry enabled locally can be helpful to debug weird errors, though.

**Reply by proko** [reply]
Hey @jamesmosier thanks for the suggestion however there is not much changing. I guess it is something related to the webpack plugin. According to [their docs](https://docs.sentry.io/platforms/javascript/guides/nextjs/manual-setup/#extend-default-webpack-usage) it should be disabled by default but at least in my case it isn't.
Whenever I run `next dev` I see the following multiple times 
```
[Sentry Webpack Plugin] Creating new release:
 'development'
[Sentry Webpack Plugin] Calling upload-sourcemaps with:
 {
  finalize: true,
  rewrite: true,
  dryRun: true,
  release: 'development',
  configFile: 'sentry.properties',
  stripPrefix: [ 'webpack://_N_E/' ],
  urlPrefix: '~/_next',
  include: [ '.next/' ],
  ignore: [
    '.next/cache',
    'server/ssr-module-cache.js',
    'static/*/_ssgManifest.js',
    'static/*/_buildManifest.js'
  ]
}
[Sentry Webpack Plugin] Finalizing release:
 'development'
```

I even tried with no custom configs at all but still get the same behaviour.

**Reply by proko** [reply]
> Sometimes, having Sentry enabled locally can be helpful to debug weird errors, though.

I totally agree @Vadorequest as long as I can disable it whenever I need or at least have an easy way to filter out dev logs from the sentry reports

**Reply by jamesmosier** [reply]
Are they actually getting uploaded to Sentry? Or are the logs just firing?

I found this from their webpack plugin, stating that they don't upload sourcemaps in development:

https://github.com/getsentry/sentry-javascript/blob/8b28573dbdd7e420cd96d8c2cd990a33ed6516d3/packages/nextjs/src/config/webpack.ts#L71
